smac/intensification/intensification.py

```diff
@@ -137,30 +192,303 @@ def __init__(self, tae_runner: ExecuteTARun,
+                    instance=instance,
+                    instance_specific=instance_specific,
+                    seed=seed,
+                    cutoff=cutoff,
+                    capped=capped,
+                    budget=0.0,
+                )
+            else:
+                raise ValueError('No valid stage found!')
+
+        except NoMoreChallengers:
+            # No new challengers are available even after a request to draw from the
+            # epm/initial configs. This means we exhausted the loop condition on line 1
+            return IntensifierBehest.TERMINATE, RunInfo(
+                config=None,
+                instance=None,
+                instance_specific="0",
+                seed=0,
+                cutoff=self.cutoff,
+                capped=False,
+                budget=0.0,
+            )
+
+    def process_results(self,
                         challenger: Configuration,
                         incumbent: typing.Optional[Configuration],
                         run_history: RunHistory,
-                        time_bound: float = float(MAXINT),
-                        log_traj: bool = True) -> typing.Tuple[Configuration, float]:
-        """Running intensification to determine the incumbent configuration.
-        *Side effect:* adds runs to run_history
+                        time_bound: float,
+                        result: RunValue,
+                        log_traj: bool = True,
+                        ) -> \
+            typing.Tuple[Configuration, float]:
+        """
+        The intensifier stage will be updated based on the results/status
+        of a configuration execution.
 
-        Implementation of Procedure 2 in Hutter et al. (2011).
+        During intensification, a the following can happen:
+        -> Challenger raced against incumbent. self.added_challenger_run
+           is set to true and that run result need to be processed
+        -> Also, during a challenger run, a capped exception
+           can be triggered, where no racer post processing is needed
+        -> A run on the incumbent for more confidence needs to
+           be processed, IntensifierStage.PROCESS_INCUMBENT_RUN
 
-        Provide either ``challengers`` or ``optimizer`` and set the other to ``None``.
-        If both arguments are given, then the optimizer will be used.
+        At the end of any run, checks are done to move to a new iteration.
 
         Parameters
         ----------
         challenger : Configuration
-            promising configurations
-        incumbent : typing.Optional[Configuration]
+            a configuration that was previously executed, and whose status
+            will be used to define the next stage.
+        incumbet : typing.Optional[Configuration]
             best configuration so far, None in 1st run
-        run_history : smac.runhistory.runhistory.RunHistory
+        run_history : typing.Optional[smac.runhistory.runhistory.RunHistory]
```

Comment:
It's not optional above, right?

Reply:
True, I fixed the docstrings accordingly.